### ISSUE TITLE
Invalid Recipients error

### ISSUE THREAD

**Issue by Seedorf161** [AppleEducate/plugins#8]
I was using the plugin but got this error on my Samsung Galaxy s8 plus running android 8.0.0
The recipients' list is formatted as `List<String>`. I think this is a bug from the plugin.

`List<String> recep = ["08106353728", "01736282629"];`
[
![screenshot_20190203-083039_messages](https://user-images.githubusercontent.com/12864618/52174424-a4cbb480-2793-11e9-971e-c2c05dbe97df.jpg)
](url)

![screenshot_20190203-083053_messages](https://user-images.githubusercontent.com/12864618/52174465-46530600-2794-11e9-9849-6bef053ccd3c.jpg)

**Comment by AppleEducate** [comment]
Which version of the plugin are you using also, please run `flutter doctor -v`

**Comment by AppleEducate** [comment]
Please test with the latest version

**Comment by nitneuq33** [comment]
same issue with galaxy s7 and 0.0.5 version

**Comment by AppleEducate** [comment]
Very weird, I will continue to test. What messaging app do you use? I am testing on android with the default one.

**Comment by nitneuq33** [comment]
I use the native message app

**Comment by AppleEducate** [comment]
Native got Samsung or android os?

**Comment by nitneuq33** [comment]
I have only one message app, after I don't know If it's an overlay of samsung or pure native

**Comment by AppleEducate** [comment]
Ok thank you, I’ll see what I can do. If you know java you are more than
welcome to make a PR

**Comment by nitneuq33** [comment]
bad news after google privacy policy change, the only solution I found to send sms is to use URL launcher, but currently some smartphone like xiaomi convert plus (+) by space ... so I'm lock I can't use your plugin because of [ ], I can't use sms 2.4.0 because It use SMS permission, I can't use URL launcher because of + who is change by space... hope somebody can help you to remove this [ ], this is my last chance

**Comment by AppleEducate** [comment]
The latest version of the plugin no longer requires the permission and uses
the updated method. I have tested it on Android 9 on a Google Pixel and
Vivo Go

**Comment by nitneuq33** [comment]
I upgrade to android 9 et now the [ ] are hide, my colleague try the exemple with xiaomi redmi note 4 and have the [ ] issue

**Comment by nitneuq33** [comment]
I found this plateform channel code for sms but need SMS permission, I don't know if this can help someone to close this issue 


First add appropriate permissions to AndroidManifest.xml.

`<uses-permission android:name="android.permission.SEND_SMS" />`
Then in your MainActivity.java:
```

package com.yourcompany.example;

import android.os.Bundle;
import android.telephony.SmsManager;
import android.util.Log;
import io.flutter.app.FlutterActivity;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugins.GeneratedPluginRegistrant;

public class MainActivity extends FlutterActivity {
  private static final String CHANNEL = "sendSms";

  private MethodChannel.Result callResult;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    GeneratedPluginRegistrant.registerWith(this);
    new MethodChannel(getFlutterView(), CHANNEL).setMethodCallHandler(
            new MethodChannel.MethodCallHandler() {
              @Override
              public void onMethodCall(MethodCall call, MethodChannel.Result result) {
                if(call.method.equals("send")){
                   String num = call.argument("phone");
                   String msg = call.argument("msg");
                   sendSMS(num,msg,result);
                }else{
                  result.notImplemented();
                }
              }
            });
  }

  private void sendSMS(String phoneNo, String msg,MethodChannel.Result result) {
      try {
          SmsManager smsManager = SmsManager.getDefault();
          smsManager.sendTextMessage(phoneNo, null, msg, null, null);
          result.success("SMS Sent");
      } catch (Exception ex) {
          ex.printStackTrace();
          result.error("Err","Sms Not Sent","");
      }
  }

}
```
Dart Code:
```

import 'dart:async';
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(new MaterialApp(
    title: "Rotation Demo",
    home: new SendSms(),
  ));
}


class SendSms extends StatefulWidget {
  @override
  _SendSmsState createState() => new _SendSmsState();
}

class _SendSmsState extends State<SendSms> {
  static const platform = const MethodChannel('sendSms');

  Future<Null> sendSms()async {
    print("SendSMS");
    try {
      final String result = await platform.invokeMethod('send',<String,dynamic>{"phone":"+91XXXXXXXXXX","msg":"Hello! I'm sent programatically."}); //Replace a 'X' with 10 digit phone number
      print(result);
    } on PlatformException catch (e) {
      print(e.toString());
    }
  }

  @override
  Widget build(BuildContext context) {
    return new Material(
      child: new Container(
        alignment: Alignment.center,
        child: new FlatButton(onPressed: () => sendSms(), child: const Text("Send SMS")),
      ),
    );
  }
}
```

**Comment by AppleEducate** [comment]
The problem is you cant add the permission any more to the manifest because you app will be removed from google play. It happened to 2 of mine. The latest version is working though

**Comment by nitneuq33** [comment]
galaxy s7 android 9 and xiaomi redmi note 4 have issues with the last version. I can't send  sms, there are recipent issue Seedorf161  has the same. So I tried to find a mix of your solution who not use permission and a solution where recipent work :)

**Comment by nitneuq33** [comment]
I found how to remove [ ] by change list string to String, but now the problem is the compilation...

I use sms 5.0.0 who need min sdk 18 and target 28

I tried to make android X migration but I have always this message :

```
D8: Program type already present: android.support.v4.app.INotificationSideChannel$Stub$Proxy

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\3.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\4.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\5.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\6.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\7.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\8.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\9.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\10.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\11.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\12.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\13.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\14.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\15.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\16.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\17.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\18.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\19.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\20.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\21.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\22.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\23.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\24.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\25.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\26.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\27.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\28.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\29.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\30.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\31.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\32.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\33.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\34.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\35.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\36.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\37.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\38.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\39.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\40.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\41.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\42.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\43.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\44.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\45.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\46.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\47.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\48.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\49.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\50.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\53.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\54.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\55.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\56.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\57.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\58.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\59.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\60.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\61.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\62.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\63.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\64.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\65.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\66.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\67.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\68.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\69.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\70.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\71.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\72.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\73.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\74.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\75.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\76.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\78.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\79.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\80.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\81.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\82.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\83.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\84.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\85.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\86.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\88.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\89.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\90.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\91.jar, C:\Users\utilisateur\AndroidStudioProjects\nirbi_final_2\build\app\intermediates\transforms\dexBuilder\debug\92.jar

```

Have you an idea ? thank you

**Comment by AppleEducate** [comment]
Android X Migration will be a pain. I ended uo in a few apps just
recreating the project and transfering the /lib

**Comment by Seedorf161** [comment]
Any fix to this error yet?

**Comment by AppleEducate** [comment]
Not yet, but If someone wants to submit a PR they can.

**Comment by AppleEducate** [comment]
Fixed with version `0.1.0`
